### PR TITLE
Add minor change for API taxons controller

### PR DESCRIPTION
**Pull request by jtapia**

### CODE REVIEW

File: api/app/controllers/spree/api/taxons_controller.rb [hunk 1]
```diff
@@ -4,21 +4,10 @@ module Spree
...
-        if taxonomy
-          @taxons = taxonomy.root.children
-        elsif params[:ids]
-          @taxons = Spree::Taxon.accessible_by(current_ability, :read).where(id: params[:ids].split(','))
-        else
-          @taxons = Spree::Taxon.accessible_by(current_ability, :read).order(:taxonomy_id, :lft).ransack(params[:q]).result
-        end
-
-        @taxons = paginate(@taxons)
+        @taxons = paginate(taxons)
```
Comment (kennyadsl) [inline api/app/controllers/spree/api/taxons_controller.rb]: It is a bit twisted for me that `@taxons` will contains the paginated version of `taxon` which sets an ivar named `@taxons` which contains the taxon scope. 

What about naming the memoizing method `taxons_scope` or `all_taxons` or something similar?

**Comment by jtapia** [comment]
I think this would make the code more readable

**Comment by jarednorman** [comment]
I'm generally not in favour of extracting methods like this without a good reason. The controller action was simple enough to begin with (not even ten actual lines of code!) and now to see how it works I have to look a hundred lines down in the file.

If someone needs to override the functionality in the extracted method or something, then the argument could be made, but without that I don't think we should merge this.

**Comment by kennyadsl** [comment]
Closing this one, there not seems to be interest, thanks for the contribution anyway.

**Pull request closed without merging**